Storage bug at JsonAdaptedLogicTaskList.java

There is a bug with out of sync Storage code. Whenever I try to load the Storage data from tasklist.json, the following error occurs:

Steps to replicate the bug:
1) Create a `data` folder in the root directory, i.e the directory containing src, docs, etc
2) Create `tasklist.json` within `data/` 

The following is the error log, in particular, I think this line is a crucial line and should be investigated first:
```at vimification.storage.JsonAdaptedLogicTaskList.toModelType(JsonAdaptedLogicTaskList.java:57)```

```
Mar 22, 2023 4:45:00 PM vimification.Gui init
INFO: =============================[ Initializing TaskPlanner ]===========================
Mar 22, 2023 4:45:00 PM vimification.Gui initConfig
INFO: Using config file : config.json
Mar 22, 2023 4:45:00 PM vimification.Gui initPrefs
INFO: Using prefs file : preferences.json
Mar 22, 2023 4:45:00 PM vimification.commons.core.LogsCenter init
INFO: currentLogLevel: INFO
Exception in Application init method
Exception in thread "main" java.lang.RuntimeException: Exception in Application init method
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:895)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:195)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NullPointerException
        at vimification.storage.JsonAdaptedLogicTaskList.toModelType(JsonAdaptedLogicTaskList.java:57)
        at vimification.storage.JsonLogicTaskListStorage.readLogicTaskList(JsonLogicTaskListStorage.java:57)
        at vimification.storage.StorageManager.readLogicTaskList(StorageManager.java:69)
        at vimification.storage.StorageManager.readLogicTaskList(StorageManager.java:62)
        at vimification.Gui.initLogicTaskList(Gui.java:78)
        at vimification.Gui.init(Gui.java:58)
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:824)
        ... 2 more
```

--- a/src/main/java/vimification/commons/util/JsonUtil.java
+++ b/src/main/java/vimification/commons/util/JsonUtil.java
@@ -53,7 +53,7 @@ public class JsonUtil {
      * Returns the Json object from the given file or {@code Optional.empty()} object if the file is
      * not found. If any values are missing from the file, default values will be used, as long as
      * the file is a valid json file.
-     * 
+     *
      * @param filePath cannot be null.
      * @param classOfObjectToDeserialize Json file has to correspond to the structure in the class
      *        given here.
@@ -62,28 +62,24 @@ public class JsonUtil {
     public static <T> Optional<T> readJsonFile(
             Path filePath, Class<T> classOfObjectToDeserialize) throws DataConversionException {
         requireNonNull(filePath);
-
         if (!Files.exists(filePath)) {
             logger.info("Json file " + filePath + " not found");
             return Optional.empty();
         }
-
-        T jsonFile;
-
+        T jsonFile = null;
         try {
             jsonFile = deserializeObjectFromJsonFile(filePath, classOfObjectToDeserialize);
         } catch (IOException e) {
             logger.warning("Error reading from jsonFile file " + filePath + ": " + e);
             throw new DataConversionException(e);
         }
-
-        return Optional.of(jsonFile);
+        return Optional.ofNullable(jsonFile);
     }
 
     /**
      * Saves the Json object to the specified file. Overwrites existing file if it exists, creates a
      * new file if it doesn't.
-     * 
+     *
      * @param jsonFile cannot be null
      * @param filePath cannot be null
      * @throws IOException if there was an error during writing to the file
@@ -98,7 +94,7 @@ public class JsonUtil {
 
     /**
      * Converts a given string representation of a JSON data to instance of a class
-     * 
+     *
      * @param <T> The generic type to create an instance of
      * @return The instance of T with the specified values in the JSON string
      */
@@ -108,7 +104,7 @@ public class JsonUtil {
 
     /**
      * Converts a given instance of a class into its JSON data string representation
-     * 
+     *
      * @param instance The T object to be converted into the JSON string
      * @param <T> The generic type to create an instance of
      * @return JSON data representation of the given class instance, in string

--- a/src/main/java/vimification/commons/util/JsonUtil.java
+++ b/src/main/java/vimification/commons/util/JsonUtil.java
@@ -69,6 +69,7 @@ public class JsonUtil {
         T jsonFile = null;
         try {
             jsonFile = deserializeObjectFromJsonFile(filePath, classOfObjectToDeserialize);
+            logger.info("Object read from json file: " + jsonFile);
         } catch (IOException e) {
             logger.warning("Error reading from jsonFile file " + filePath + ": " + e);
             throw new DataConversionException(e);

--- a/src/main/java/vimification/storage/JsonAdaptedLogicTaskList.java
+++ b/src/main/java/vimification/storage/JsonAdaptedLogicTaskList.java
@@ -28,11 +28,7 @@ public class JsonAdaptedLogicTaskList {
 
     @JsonCreator
     public JsonAdaptedLogicTaskList(@JsonProperty("tasks") List<JsonAdaptedTask> tasks) {
-        if (tasks == null) {
-            LOGGER.warning("JsonAdaptedLogicTaskList: task list read from file is null");
-            tasks = List.of();
-        }
-        this.tasks = tasks;
+        this.tasks = tasks == null ? List.of() : tasks;
     }
 
     /**
@@ -67,5 +63,10 @@ public class JsonAdaptedLogicTaskList {
             taskList.add(task);
         }
         return taskList;
+    }
+
+    @Override
+    public String toString() {
+        return "JsonAdpatedLogicTaskList: [" + tasks + "]";
     }
 }

--- a/src/main/java/vimification/storage/JsonAdaptedLogicTaskList.java
+++ b/src/main/java/vimification/storage/JsonAdaptedLogicTaskList.java
@@ -1,12 +1,14 @@
 package vimification.storage;
 
 import java.util.List;
+import java.util.logging.Logger;
 import java.util.stream.Collectors;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonRootName;
 
+import vimification.commons.core.LogsCenter;
 import vimification.commons.exceptions.IllegalValueException;
 import vimification.model.LogicTaskList;
 import vimification.model.task.Deadline;
@@ -20,10 +22,16 @@ import vimification.model.task.Todo;
 @JsonRootName(value = "logicTaskList")
 public class JsonAdaptedLogicTaskList {
 
+    private static final Logger LOGGER = LogsCenter.getLogger(JsonAdaptedLogicTaskList.class);
+
     private final List<JsonAdaptedTask> tasks;
 
     @JsonCreator
     public JsonAdaptedLogicTaskList(@JsonProperty("tasks") List<JsonAdaptedTask> tasks) {
+        if (tasks == null) {
+            LOGGER.warning("JsonAdaptedLogicTaskList: task list read from file is null");
+            tasks = List.of();
+        }
         this.tasks = tasks;
     }
 

--- a/src/main/java/vimification/storage/JsonLogicTaskListStorage.java
+++ b/src/main/java/vimification/storage/JsonLogicTaskListStorage.java
@@ -19,7 +19,7 @@ import vimification.model.LogicTaskList;
  */
 public class JsonLogicTaskListStorage implements LogicTaskListStorage {
 
-    private static final Logger logger = LogsCenter.getLogger(JsonLogicTaskListStorage.class);
+    private static final Logger LOGGER = LogsCenter.getLogger(JsonLogicTaskListStorage.class);
 
     private Path filePath;
 
@@ -49,6 +49,7 @@ public class JsonLogicTaskListStorage implements LogicTaskListStorage {
 
         Optional<JsonAdaptedLogicTaskList> jsonTaskList = JsonUtil.readJsonFile(
                 filePath, JsonAdaptedLogicTaskList.class);
+        LOGGER.info(jsonTaskList.toString());
         if (!jsonTaskList.isPresent()) {
             return Optional.empty();
         }
@@ -56,7 +57,7 @@ public class JsonLogicTaskListStorage implements LogicTaskListStorage {
         try {
             return Optional.of(jsonTaskList.get().toModelType());
         } catch (IllegalValueException ive) {
-            logger.info("Illegal values found in " + filePath + ": " + ive.getMessage());
+            LOGGER.info("Illegal values found in " + filePath + ": " + ive.getMessage());
             throw new DataConversionException(ive);
         }
     }

